clean/new_orleans_pd_stop_and_search.py

```diff
@@ -0,0 +1,236 @@
+    col = df.actionstaken.str.lower().str.strip()\
+        .str.extract(r'(?:(consent to search\: (\w+)))')
+
+    df.loc[:, 'consent_to_search'] = col[0]\
+        .str.replace(r'consent to search\: ', '', regex=True)
+    return df
+
+
+def extract_consent_form_column(df):
+    col = df.actionstaken.str.lower().str.strip()\
+        .str.extract(r'(?:(consent form completed\: (\w+)))')
+
+    df.loc[:, 'consent_form_completed'] = col[0]\
+        .str.replace(r'consent form completed\: ', '', regex=True)
+    return df
+
+
+def extract_body_search_column(df):
+    col = df.actionstaken.str.lower().str.strip()\
+        .str.extract(r'(?:(stripbody cavity search\: (\w+)))')
+
+    df.loc[:, 'strip_body_cavity_search'] = col[0]\
+        .str.replace(r'stripbody cavity search\: ', '', regex=True)
+    return df
+
+
+def extract_exit_vehicle_column(df):
+    col = df.actionstaken.str.lower().str.strip()\
+        .str.extract(r'(?:(exit vehicle\: (\w+)))')
+
+    df.loc[:, 'exit_vehicle'] = col[0]\
+        .str.replace(r'exit vehicle\: ', '', regex=True)
+    return df.drop(columns='actionstaken')
+
+
+def clean_assigned_department(df):
+    df.loc[:, 'assigned_department'] = df.officerassignment.str.lower().str.strip()\
+        .str.replace('pib', 'public integrity bureau', regex=False)\
+        .str.replace('fob', 'field operations bureau', regex=False)\
+        .str.replace('msb', 'management services bureau', regex=False)\
+        .str.replace('superintendent', 'office of the superintendent', regex=False)\
+        .str.replace('isb', 'investigations and support bureau', regex=False)\
+        .str.replace(r'((.+) district)', '', regex=True)
+    return df.drop(columns='officerassignment')
+
+
+def consolidate_name_and_badge_columns(df):
+    df.loc[:, 'officer1badgenumber'] = df.officer1badgenumber.fillna('').astype(str)\
+        .str.replace(r'^(\w+)$', r'(\1)', regex=True)
+    df.loc[:, 'officer2badgenumber'] = df.officer2badgenumber.fillna('').astype(str)\
+        .str.replace(r'^(\w+)\.(\w+)$', r'(\1)', regex=True)
+
+    df.loc[:, 'officer1name'] = df.officer1name.str.cat(df.officer1badgenumber, sep=' ')
+    df.loc[:, 'officer2name'] = df.officer2name.fillna('').str.cat(df.officer2badgenumber, sep=' ')
+
+    df.loc[:, 'officer_names_and_badges'] = df.officer1name + '/' + df.officer2name
+    df.loc[:, 'officer_names_and_badges'] = df.officer_names_and_badges.str.lower().str.strip()\
+        .str.replace(r'\/$', '', regex=True)\
+        .str.replace(r'(\w+)\/(\w+) ', r'\1.\2', regex=True)
+    return df.drop(columns=['officer1badgenumber', 'officer2badgenumber', 'officer1name', 'officer2name'])
+
+
+def split_rows_with_multiple_officers(df):
+    df = df.drop('officer_names_and_badges', axis=1).join(df['officer_names_and_badges']\
+        .str.split('/', expand=True)\
+        .stack()\
+        .reset_index(level=1, drop=True).rename('officer_names_and_badges'))\
+        .reset_index(drop=True)
```

Comment:
Have you checked how many rows the final frame has? You might need to add `how='outer'` to `join` method. Also there's no need for backslash within brackets.

Reply:
Thanks. The final frame has 830301 rows.

Reply:
Very nice change btw!

Reply:
Awesome. Thanks. Converting to draft to complete matching and fuse.